// @adyen/api-library/dist/lib-esm/src/typings/recurring/name.js
/**
 * Adyen Recurring API
 * The Recurring APIs allow you to manage and remove your tokens or saved payment details. Tokens should be created with validation during a payment request.  For more information, refer to our [Tokenization documentation](https://docs.adyen.com/checkout/tokenization). ## Authentication To connect to the Recurring API, you must use your basic authentication credentials. For this, create your web service user, as described in [How to get the WS user password](https://docs.adyen.com/user-management/how-to-get-the-web-service-ws-user-password). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@Company.YourCompany\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning Recurring API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://pal-test.adyen.com/pal/servlet/Recurring/v49/disable ```
 *
 * The version of the OpenAPI document: 49
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var Name = /** @class */ (function () {
    function Name() {
    }
    Name.getAttributeTypeMap = function () {
        return Name.attributeTypeMap;
    };
    Name.discriminator = undefined;
    Name.attributeTypeMap = [
        {
            "name": "firstName",
            "baseName": "firstName",
            "type": "string"
        },
        {
            "name": "gender",
            "baseName": "gender",
            "type": "Name.GenderEnum"
        },
        {
            "name": "infix",
            "baseName": "infix",
            "type": "string"
        },
        {
            "name": "lastName",
            "baseName": "lastName",
            "type": "string"
        }
    ];
    return Name;
}());
export { Name };
(function (Name) {
    var GenderEnum;
    (function (GenderEnum) {
        GenderEnum[GenderEnum["MALE"] = 'MALE'] = "MALE";
        GenderEnum[GenderEnum["FEMALE"] = 'FEMALE'] = "FEMALE";
        GenderEnum[GenderEnum["UNKNOWN"] = 'UNKNOWN'] = "UNKNOWN";
    })(GenderEnum = Name.GenderEnum || (Name.GenderEnum = {}));
})(Name || (Name = {}));
