// @adyen/api-library/dist/lib-esm/src/typings/recurring/recurring.js
/**
 * Adyen Recurring API
 * The Recurring APIs allow you to manage and remove your tokens or saved payment details. Tokens should be created with validation during a payment request.  For more information, refer to our [Tokenization documentation](https://docs.adyen.com/checkout/tokenization). ## Authentication To connect to the Recurring API, you must use your basic authentication credentials. For this, create your web service user, as described in [How to get the WS user password](https://docs.adyen.com/user-management/how-to-get-the-web-service-ws-user-password). Then use its credentials to authenticate your request, for example:  ``` curl -U \"ws@Company.YourCompany\":\"YourWsPassword\" \\ -H \"Content-Type: application/json\" \\ ... ``` Note that when going live, you need to generate new web service user credentials to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning Recurring API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://pal-test.adyen.com/pal/servlet/Recurring/v49/disable ```
 *
 * The version of the OpenAPI document: 49
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var Recurring = /** @class */ (function () {
    function Recurring() {
    }
    Recurring.getAttributeTypeMap = function () {
        return Recurring.attributeTypeMap;
    };
    Recurring.discriminator = undefined;
    Recurring.attributeTypeMap = [
        {
            "name": "contract",
            "baseName": "contract",
            "type": "Recurring.ContractEnum"
        },
        {
            "name": "recurringDetailName",
            "baseName": "recurringDetailName",
            "type": "string"
        },
        {
            "name": "recurringExpiry",
            "baseName": "recurringExpiry",
            "type": "Date"
        },
        {
            "name": "recurringFrequency",
            "baseName": "recurringFrequency",
            "type": "string"
        },
        {
            "name": "tokenService",
            "baseName": "tokenService",
            "type": "Recurring.TokenServiceEnum"
        }
    ];
    return Recurring;
}());
export { Recurring };
(function (Recurring) {
    var ContractEnum;
    (function (ContractEnum) {
        ContractEnum[ContractEnum["ONECLICK"] = 'ONECLICK'] = "ONECLICK";
        ContractEnum[ContractEnum["RECURRING"] = 'RECURRING'] = "RECURRING";
        ContractEnum[ContractEnum["PAYOUT"] = 'PAYOUT'] = "PAYOUT";
    })(ContractEnum = Recurring.ContractEnum || (Recurring.ContractEnum = {}));
    var TokenServiceEnum;
    (function (TokenServiceEnum) {
        TokenServiceEnum[TokenServiceEnum["VISATOKENSERVICE"] = 'VISATOKENSERVICE'] = "VISATOKENSERVICE";
        TokenServiceEnum[TokenServiceEnum["MCTOKENSERVICE"] = 'MCTOKENSERVICE'] = "MCTOKENSERVICE";
    })(TokenServiceEnum = Recurring.TokenServiceEnum || (Recurring.TokenServiceEnum = {}));
})(Recurring || (Recurring = {}));
