// @adyen/api-library/dist/lib/src/typings/binLookup/recurring.js
"use strict";
/**
 * Adyen BinLookup API
 * The BIN Lookup API provides endpoints for retrieving information, such as cost estimates, and 3D Secure supported version based on a given BIN.
 *
 * The version of the OpenAPI document: 50
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var Recurring = /** @class */ (function () {
    function Recurring() {
    }
    Recurring.getAttributeTypeMap = function () {
        return Recurring.attributeTypeMap;
    };
    Recurring.discriminator = undefined;
    Recurring.attributeTypeMap = [
        {
            "name": "contract",
            "baseName": "contract",
            "type": "Recurring.ContractEnum"
        },
        {
            "name": "recurringDetailName",
            "baseName": "recurringDetailName",
            "type": "string"
        },
        {
            "name": "recurringExpiry",
            "baseName": "recurringExpiry",
            "type": "Date"
        },
        {
            "name": "recurringFrequency",
            "baseName": "recurringFrequency",
            "type": "string"
        },
        {
            "name": "tokenService",
            "baseName": "tokenService",
            "type": "Recurring.TokenServiceEnum"
        }
    ];
    return Recurring;
}());
exports.Recurring = Recurring;
(function (Recurring) {
    var ContractEnum;
    (function (ContractEnum) {
        ContractEnum[ContractEnum["ONECLICK"] = 'ONECLICK'] = "ONECLICK";
        ContractEnum[ContractEnum["RECURRING"] = 'RECURRING'] = "RECURRING";
        ContractEnum[ContractEnum["PAYOUT"] = 'PAYOUT'] = "PAYOUT";
    })(ContractEnum = Recurring.ContractEnum || (Recurring.ContractEnum = {}));
    var TokenServiceEnum;
    (function (TokenServiceEnum) {
        TokenServiceEnum[TokenServiceEnum["VISATOKENSERVICE"] = 'VISATOKENSERVICE'] = "VISATOKENSERVICE";
        TokenServiceEnum[TokenServiceEnum["MCTOKENSERVICE"] = 'MCTOKENSERVICE'] = "MCTOKENSERVICE";
    })(TokenServiceEnum = Recurring.TokenServiceEnum || (Recurring.TokenServiceEnum = {}));
})(Recurring = exports.Recurring || (exports.Recurring = {}));
exports.Recurring = Recurring;
