// @adyen/api-library/dist/lib-esm/src/typings/binLookup/costEstimateRequest.js
/**
 * Adyen BinLookup API
 * The BIN Lookup API provides endpoints for retrieving information, such as cost estimates, and 3D Secure supported version based on a given BIN.
 *
 * The version of the OpenAPI document: 50
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var CostEstimateRequest = /** @class */ (function () {
    function CostEstimateRequest() {
    }
    CostEstimateRequest.getAttributeTypeMap = function () {
        return CostEstimateRequest.attributeTypeMap;
    };
    CostEstimateRequest.discriminator = undefined;
    CostEstimateRequest.attributeTypeMap = [
        {
            "name": "amount",
            "baseName": "amount",
            "type": "Amount"
        },
        {
            "name": "assumptions",
            "baseName": "assumptions",
            "type": "CostEstimateAssumptions"
        },
        {
            "name": "cardNumber",
            "baseName": "cardNumber",
            "type": "string"
        },
        {
            "name": "encryptedCard",
            "baseName": "encryptedCard",
            "type": "string"
        },
        {
            "name": "merchantAccount",
            "baseName": "merchantAccount",
            "type": "string"
        },
        {
            "name": "merchantDetails",
            "baseName": "merchantDetails",
            "type": "MerchantDetails"
        },
        {
            "name": "recurring",
            "baseName": "recurring",
            "type": "Recurring"
        },
        {
            "name": "selectedRecurringDetailReference",
            "baseName": "selectedRecurringDetailReference",
            "type": "string"
        },
        {
            "name": "shopperInteraction",
            "baseName": "shopperInteraction",
            "type": "CostEstimateRequest.ShopperInteractionEnum"
        },
        {
            "name": "shopperReference",
            "baseName": "shopperReference",
            "type": "string"
        }
    ];
    return CostEstimateRequest;
}());
export { CostEstimateRequest };
(function (CostEstimateRequest) {
    var ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum[ShopperInteractionEnum["Ecommerce"] = 'Ecommerce'] = "Ecommerce";
        ShopperInteractionEnum[ShopperInteractionEnum["ContAuth"] = 'ContAuth'] = "ContAuth";
        ShopperInteractionEnum[ShopperInteractionEnum["Moto"] = 'Moto'] = "Moto";
        ShopperInteractionEnum[ShopperInteractionEnum["POS"] = 'POS'] = "POS";
    })(ShopperInteractionEnum = CostEstimateRequest.ShopperInteractionEnum || (CostEstimateRequest.ShopperInteractionEnum = {}));
})(CostEstimateRequest || (CostEstimateRequest = {}));
