// @adyen/api-library/dist/lib/src/typings/payout/deviceRenderOptions.js
"use strict";
/**
 * Adyen Payout API
 * A set of API endpoints that allow you to store payout details, confirm, or decline a payout.  For more information, refer to [Online payouts](https://docs.adyen.com/checkout/online-payouts).
 *
 * The version of the OpenAPI document: 51
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var DeviceRenderOptions = /** @class */ (function () {
    function DeviceRenderOptions() {
    }
    DeviceRenderOptions.getAttributeTypeMap = function () {
        return DeviceRenderOptions.attributeTypeMap;
    };
    DeviceRenderOptions.discriminator = undefined;
    DeviceRenderOptions.attributeTypeMap = [
        {
            "name": "sdkInterface",
            "baseName": "sdkInterface",
            "type": "DeviceRenderOptions.SdkInterfaceEnum"
        },
        {
            "name": "sdkUiType",
            "baseName": "sdkUiType",
            "type": "Array<DeviceRenderOptions.SdkUiTypeEnum>"
        }
    ];
    return DeviceRenderOptions;
}());
exports.DeviceRenderOptions = DeviceRenderOptions;
(function (DeviceRenderOptions) {
    var SdkInterfaceEnum;
    (function (SdkInterfaceEnum) {
        SdkInterfaceEnum[SdkInterfaceEnum["Html"] = 'Html'] = "Html";
        SdkInterfaceEnum[SdkInterfaceEnum["Native"] = 'Native'] = "Native";
        SdkInterfaceEnum[SdkInterfaceEnum["Both"] = 'both'] = "Both";
    })(SdkInterfaceEnum = DeviceRenderOptions.SdkInterfaceEnum || (DeviceRenderOptions.SdkInterfaceEnum = {}));
    var SdkUiTypeEnum;
    (function (SdkUiTypeEnum) {
        SdkUiTypeEnum[SdkUiTypeEnum["MultiSelect"] = 'multiSelect'] = "MultiSelect";
        SdkUiTypeEnum[SdkUiTypeEnum["OtherHtml"] = 'otherHtml'] = "OtherHtml";
        SdkUiTypeEnum[SdkUiTypeEnum["OutOfBand"] = 'outOfBand'] = "OutOfBand";
        SdkUiTypeEnum[SdkUiTypeEnum["SingleSelect"] = 'singleSelect'] = "SingleSelect";
        SdkUiTypeEnum[SdkUiTypeEnum["Text"] = 'text'] = "Text";
    })(SdkUiTypeEnum = DeviceRenderOptions.SdkUiTypeEnum || (DeviceRenderOptions.SdkUiTypeEnum = {}));
})(DeviceRenderOptions = exports.DeviceRenderOptions || (exports.DeviceRenderOptions = {}));
exports.DeviceRenderOptions = DeviceRenderOptions;
