// @adyen/api-library/dist/lib/src/typings/payout/threeDSecureData.js
"use strict";
/**
 * Adyen Payout API
 * A set of API endpoints that allow you to store payout details, confirm, or decline a payout.  For more information, refer to [Online payouts](https://docs.adyen.com/checkout/online-payouts).
 *
 * The version of the OpenAPI document: 51
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var ThreeDSecureData = /** @class */ (function () {
    function ThreeDSecureData() {
    }
    ThreeDSecureData.getAttributeTypeMap = function () {
        return ThreeDSecureData.attributeTypeMap;
    };
    ThreeDSecureData.discriminator = undefined;
    ThreeDSecureData.attributeTypeMap = [
        {
            "name": "authenticationResponse",
            "baseName": "authenticationResponse",
            "type": "ThreeDSecureData.AuthenticationResponseEnum"
        },
        {
            "name": "cavv",
            "baseName": "cavv",
            "type": "string"
        },
        {
            "name": "cavvAlgorithm",
            "baseName": "cavvAlgorithm",
            "type": "string"
        },
        {
            "name": "directoryResponse",
            "baseName": "directoryResponse",
            "type": "ThreeDSecureData.DirectoryResponseEnum"
        },
        {
            "name": "dsTransID",
            "baseName": "dsTransID",
            "type": "string"
        },
        {
            "name": "eci",
            "baseName": "eci",
            "type": "string"
        },
        {
            "name": "threeDSVersion",
            "baseName": "threeDSVersion",
            "type": "string"
        },
        {
            "name": "xid",
            "baseName": "xid",
            "type": "string"
        }
    ];
    return ThreeDSecureData;
}());
exports.ThreeDSecureData = ThreeDSecureData;
(function (ThreeDSecureData) {
    var AuthenticationResponseEnum;
    (function (AuthenticationResponseEnum) {
        AuthenticationResponseEnum[AuthenticationResponseEnum["Y"] = 'Y'] = "Y";
        AuthenticationResponseEnum[AuthenticationResponseEnum["N"] = 'N'] = "N";
        AuthenticationResponseEnum[AuthenticationResponseEnum["U"] = 'U'] = "U";
        AuthenticationResponseEnum[AuthenticationResponseEnum["A"] = 'A'] = "A";
    })(AuthenticationResponseEnum = ThreeDSecureData.AuthenticationResponseEnum || (ThreeDSecureData.AuthenticationResponseEnum = {}));
    var DirectoryResponseEnum;
    (function (DirectoryResponseEnum) {
        DirectoryResponseEnum[DirectoryResponseEnum["A"] = 'A'] = "A";
        DirectoryResponseEnum[DirectoryResponseEnum["C"] = 'C'] = "C";
        DirectoryResponseEnum[DirectoryResponseEnum["D"] = 'D'] = "D";
        DirectoryResponseEnum[DirectoryResponseEnum["I"] = 'I'] = "I";
        DirectoryResponseEnum[DirectoryResponseEnum["N"] = 'N'] = "N";
        DirectoryResponseEnum[DirectoryResponseEnum["R"] = 'R'] = "R";
        DirectoryResponseEnum[DirectoryResponseEnum["U"] = 'U'] = "U";
        DirectoryResponseEnum[DirectoryResponseEnum["Y"] = 'Y'] = "Y";
    })(DirectoryResponseEnum = ThreeDSecureData.DirectoryResponseEnum || (ThreeDSecureData.DirectoryResponseEnum = {}));
})(ThreeDSecureData = exports.ThreeDSecureData || (exports.ThreeDSecureData = {}));
exports.ThreeDSecureData = ThreeDSecureData;
