// @adyen/api-library/dist/lib-esm/src/typings/payout/merchantRiskIndicator.js
/**
 * Adyen Payout API
 * A set of API endpoints that allow you to store payout details, confirm, or decline a payout.  For more information, refer to [Online payouts](https://docs.adyen.com/checkout/online-payouts).
 *
 * The version of the OpenAPI document: 51
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var MerchantRiskIndicator = /** @class */ (function () {
    function MerchantRiskIndicator() {
    }
    MerchantRiskIndicator.getAttributeTypeMap = function () {
        return MerchantRiskIndicator.attributeTypeMap;
    };
    MerchantRiskIndicator.discriminator = undefined;
    MerchantRiskIndicator.attributeTypeMap = [
        {
            "name": "addressMatch",
            "baseName": "addressMatch",
            "type": "boolean"
        },
        {
            "name": "deliveryAddressIndicator",
            "baseName": "deliveryAddressIndicator",
            "type": "MerchantRiskIndicator.DeliveryAddressIndicatorEnum"
        },
        {
            "name": "deliveryEmail",
            "baseName": "deliveryEmail",
            "type": "string"
        },
        {
            "name": "deliveryTimeframe",
            "baseName": "deliveryTimeframe",
            "type": "MerchantRiskIndicator.DeliveryTimeframeEnum"
        },
        {
            "name": "giftCardAmount",
            "baseName": "giftCardAmount",
            "type": "Amount"
        },
        {
            "name": "giftCardCount",
            "baseName": "giftCardCount",
            "type": "number"
        },
        {
            "name": "preOrderDate",
            "baseName": "preOrderDate",
            "type": "Date"
        },
        {
            "name": "preOrderPurchase",
            "baseName": "preOrderPurchase",
            "type": "boolean"
        },
        {
            "name": "reorderItems",
            "baseName": "reorderItems",
            "type": "boolean"
        }
    ];
    return MerchantRiskIndicator;
}());
export { MerchantRiskIndicator };
(function (MerchantRiskIndicator) {
    var DeliveryAddressIndicatorEnum;
    (function (DeliveryAddressIndicatorEnum) {
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["ShipToBillingAddress"] = 'shipToBillingAddress'] = "ShipToBillingAddress";
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["ShipToVerifiedAddress"] = 'shipToVerifiedAddress'] = "ShipToVerifiedAddress";
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["ShipToNewAddress"] = 'shipToNewAddress'] = "ShipToNewAddress";
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["ShipToStore"] = 'shipToStore'] = "ShipToStore";
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["DigitalGoods"] = 'digitalGoods'] = "DigitalGoods";
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["GoodsNotShipped"] = 'goodsNotShipped'] = "GoodsNotShipped";
        DeliveryAddressIndicatorEnum[DeliveryAddressIndicatorEnum["Other"] = 'other'] = "Other";
    })(DeliveryAddressIndicatorEnum = MerchantRiskIndicator.DeliveryAddressIndicatorEnum || (MerchantRiskIndicator.DeliveryAddressIndicatorEnum = {}));
    var DeliveryTimeframeEnum;
    (function (DeliveryTimeframeEnum) {
        DeliveryTimeframeEnum[DeliveryTimeframeEnum["ElectronicDelivery"] = 'electronicDelivery'] = "ElectronicDelivery";
        DeliveryTimeframeEnum[DeliveryTimeframeEnum["SameDayShipping"] = 'sameDayShipping'] = "SameDayShipping";
        DeliveryTimeframeEnum[DeliveryTimeframeEnum["OvernightShipping"] = 'overnightShipping'] = "OvernightShipping";
        DeliveryTimeframeEnum[DeliveryTimeframeEnum["TwoOrMoreDaysShipping"] = 'twoOrMoreDaysShipping'] = "TwoOrMoreDaysShipping";
    })(DeliveryTimeframeEnum = MerchantRiskIndicator.DeliveryTimeframeEnum || (MerchantRiskIndicator.DeliveryTimeframeEnum = {}));
})(MerchantRiskIndicator || (MerchantRiskIndicator = {}));
