// @adyen/api-library/dist/lib-esm/src/typings/payout/name.js
/**
 * Adyen Payout API
 * A set of API endpoints that allow you to store payout details, confirm, or decline a payout.  For more information, refer to [Online payouts](https://docs.adyen.com/checkout/online-payouts).
 *
 * The version of the OpenAPI document: 51
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var Name = /** @class */ (function () {
    function Name() {
    }
    Name.getAttributeTypeMap = function () {
        return Name.attributeTypeMap;
    };
    Name.discriminator = undefined;
    Name.attributeTypeMap = [
        {
            "name": "firstName",
            "baseName": "firstName",
            "type": "string"
        },
        {
            "name": "gender",
            "baseName": "gender",
            "type": "Name.GenderEnum"
        },
        {
            "name": "infix",
            "baseName": "infix",
            "type": "string"
        },
        {
            "name": "lastName",
            "baseName": "lastName",
            "type": "string"
        }
    ];
    return Name;
}());
export { Name };
(function (Name) {
    var GenderEnum;
    (function (GenderEnum) {
        GenderEnum[GenderEnum["MALE"] = 'MALE'] = "MALE";
        GenderEnum[GenderEnum["FEMALE"] = 'FEMALE'] = "FEMALE";
        GenderEnum[GenderEnum["UNKNOWN"] = 'UNKNOWN'] = "UNKNOWN";
    })(GenderEnum = Name.GenderEnum || (Name.GenderEnum = {}));
})(Name || (Name = {}));
