// @adyen/api-library/dist/lib-esm/src/typings/payout/payoutResponse.js
/**
 * Adyen Payout API
 * A set of API endpoints that allow you to store payout details, confirm, or decline a payout.  For more information, refer to [Online payouts](https://docs.adyen.com/checkout/online-payouts).
 *
 * The version of the OpenAPI document: 51
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var PayoutResponse = /** @class */ (function () {
    function PayoutResponse() {
    }
    PayoutResponse.getAttributeTypeMap = function () {
        return PayoutResponse.attributeTypeMap;
    };
    PayoutResponse.discriminator = undefined;
    PayoutResponse.attributeTypeMap = [
        {
            "name": "additionalData",
            "baseName": "additionalData",
            "type": "object"
        },
        {
            "name": "authCode",
            "baseName": "authCode",
            "type": "string"
        },
        {
            "name": "dccAmount",
            "baseName": "dccAmount",
            "type": "Amount"
        },
        {
            "name": "dccSignature",
            "baseName": "dccSignature",
            "type": "string"
        },
        {
            "name": "fraudResult",
            "baseName": "fraudResult",
            "type": "FraudResult"
        },
        {
            "name": "issuerUrl",
            "baseName": "issuerUrl",
            "type": "string"
        },
        {
            "name": "md",
            "baseName": "md",
            "type": "string"
        },
        {
            "name": "paRequest",
            "baseName": "paRequest",
            "type": "string"
        },
        {
            "name": "pspReference",
            "baseName": "pspReference",
            "type": "string"
        },
        {
            "name": "refusalReason",
            "baseName": "refusalReason",
            "type": "string"
        },
        {
            "name": "resultCode",
            "baseName": "resultCode",
            "type": "PayoutResponse.ResultCodeEnum"
        }
    ];
    return PayoutResponse;
}());
export { PayoutResponse };
(function (PayoutResponse) {
    var ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum[ResultCodeEnum["AuthenticationFinished"] = 'AuthenticationFinished'] = "AuthenticationFinished";
        ResultCodeEnum[ResultCodeEnum["Authorised"] = 'Authorised'] = "Authorised";
        ResultCodeEnum[ResultCodeEnum["Cancelled"] = 'Cancelled'] = "Cancelled";
        ResultCodeEnum[ResultCodeEnum["ChallengeShopper"] = 'ChallengeShopper'] = "ChallengeShopper";
        ResultCodeEnum[ResultCodeEnum["Error"] = 'Error'] = "Error";
        ResultCodeEnum[ResultCodeEnum["IdentifyShopper"] = 'IdentifyShopper'] = "IdentifyShopper";
        ResultCodeEnum[ResultCodeEnum["Pending"] = 'Pending'] = "Pending";
        ResultCodeEnum[ResultCodeEnum["PresentToShopper"] = 'PresentToShopper'] = "PresentToShopper";
        ResultCodeEnum[ResultCodeEnum["Received"] = 'Received'] = "Received";
        ResultCodeEnum[ResultCodeEnum["RedirectShopper"] = 'RedirectShopper'] = "RedirectShopper";
        ResultCodeEnum[ResultCodeEnum["Refused"] = 'Refused'] = "Refused";
    })(ResultCodeEnum = PayoutResponse.ResultCodeEnum || (PayoutResponse.ResultCodeEnum = {}));
})(PayoutResponse || (PayoutResponse = {}));
