// @adyen/api-library/dist/lib-esm/src/typings/payout/split.js
/**
 * Adyen Payout API
 * A set of API endpoints that allow you to store payout details, confirm, or decline a payout.  For more information, refer to [Online payouts](https://docs.adyen.com/checkout/online-payouts).
 *
 * The version of the OpenAPI document: 51
 * Contact: support@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var Split = /** @class */ (function () {
    function Split() {
    }
    Split.getAttributeTypeMap = function () {
        return Split.attributeTypeMap;
    };
    Split.discriminator = undefined;
    Split.attributeTypeMap = [
        {
            "name": "account",
            "baseName": "account",
            "type": "string"
        },
        {
            "name": "amount",
            "baseName": "amount",
            "type": "SplitAmount"
        },
        {
            "name": "description",
            "baseName": "description",
            "type": "string"
        },
        {
            "name": "reference",
            "baseName": "reference",
            "type": "string"
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "Split.TypeEnum"
        }
    ];
    return Split;
}());
export { Split };
(function (Split) {
    var TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Commission"] = 'Commission'] = "Commission";
        TypeEnum[TypeEnum["Default"] = 'Default'] = "Default";
        TypeEnum[TypeEnum["MarketPlace"] = 'MarketPlace'] = "MarketPlace";
        TypeEnum[TypeEnum["PaymentFee"] = 'PaymentFee'] = "PaymentFee";
        TypeEnum[TypeEnum["VAT"] = 'VAT'] = "VAT";
        TypeEnum[TypeEnum["Verification"] = 'Verification'] = "Verification";
    })(TypeEnum = Split.TypeEnum || (Split.TypeEnum = {}));
})(Split || (Split = {}));
